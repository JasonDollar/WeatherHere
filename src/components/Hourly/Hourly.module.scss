.Summary {
  font-size: 2.5rem;
  text-align: center;
  margin: 2rem 0 1rem 0;
}


@media only screen and (max-width: 576px) {
  .chartContainer {
    overflow-x: hidden;
    margin: 0 1rem 0 -3rem;
  }

}