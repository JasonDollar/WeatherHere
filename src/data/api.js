const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

