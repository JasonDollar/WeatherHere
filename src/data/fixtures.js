import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}