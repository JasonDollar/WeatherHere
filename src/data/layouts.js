const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



