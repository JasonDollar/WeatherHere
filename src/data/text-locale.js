const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 