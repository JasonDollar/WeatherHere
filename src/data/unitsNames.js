export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}