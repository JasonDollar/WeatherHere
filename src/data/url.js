const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }