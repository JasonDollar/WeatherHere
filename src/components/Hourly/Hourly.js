import React from 'react'
import { 
  ResponsiveContainer, LineChart, XAxis, Line, YAxis, CartesianGrid, Tooltip, Legend,
} from 'recharts'
import { getTimeFromSeconds, formatNumber } from '../../data/utils'

import CustomTooltip from '../CustomTooltip/CustomTooltip'

import classes from './Hourly.module.scss'

const Hourly = ({ hourly, hourlyText, timezone }) => { 

  const data = hourly.data
    .filter((item, index) => {
      if (index === 0 || index % 3 === 0) {
        return true
      } 
      return false
    })
    .map(item => {
      const precipProb = item.precipProbability > 0.05 ? item.precipProbability : 0
      const time = getTimeFromSeconds(item.time, timezone)
      return {
        [hourlyText.temperature]: formatNumber(item.temperature),
        [hourlyText.precipProb]: formatNumber(precipProb * 100),
        summary: item.summary,
        time: time.hour,
      }
    })

  const style = '#dd0055'
  return (
    <div>
      <p className={classes.Summary}>{hourly.summary}</p>
      <ResponsiveContainer width="98%" height={300}>
        <LineChart data={data}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="time" />
          <YAxis yAxisId="left" label="Â°C" type="number" tick={{ stroke: style }} domain={['auto', 'auto']} />
          <YAxis yAxisId="right" orientation="right" label="%" type="number" domain={[0, 'dataMax + 20']} />
          <Tooltip
            content={<CustomTooltip tempText={hourlyText} type="hourly" />} 
            wrapperStyle={{
              backgroundColor: '#fff',
              border: '1px solid #555',
              borderRadius: '5px',
              boxShadow: '0 0 20px -5px rgb(18, 18, 18)',
            }} 
          />
          <Legend />
          <Line yAxisId="left" type="monotone" dataKey={hourlyText.temperature} stroke={style} activeDot={{ r: 8 }} />
          <Line yAxisId="right" type="monotone" dataKey={hourlyText.precipProb} stroke="#82ca9d" />
        </LineChart>
      </ResponsiveContainer>
      
    </div>
  )
}

export default Hourly

