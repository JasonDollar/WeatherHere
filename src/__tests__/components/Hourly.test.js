import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import Hourly from '../../components/Hourly/Hourly'

let hourly
let hourlyText
let timezone
let units
let wrapper

beforeAll(() => {
  hourly = data.weather.hourly
  hourlyText = data.text.pl.date
  timezone = data.timezone
  units = data.units.si.id
  wrapper = shallow(
    <Hourly 
      hourly={hourly}
      hourlyText={hourlyText}
      timezone={timezone}
      units={units}
    />,
  )
})

describe('Hourly component test', () => {
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('proper props given to Summary component', () => {
    const element = wrapper.find('Summary')
    expect(element.props().children).toBe(hourly.summary) 
  }) 
})