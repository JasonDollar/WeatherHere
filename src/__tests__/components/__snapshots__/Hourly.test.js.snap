// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hourly component test Matches snapshot 1`] = `
<section
  className="Hourly"
>
  <Summary>
    Duże zachmurzenie w ciągu dnia, nocą słaby wiatr, skończy się jutro rano.
  </Summary>
  <div
    className="chartContainer"
  >
    <ResponsiveContainer
      debounce={0}
      height={300}
      width="100%"
    >
      <LineChart
        barCategoryGap="10%"
        barGap={4}
        data={
          Array [
            Object {
              "Temperatura (°C)": 9,
              "summary": "Średnie zachmurzenie",
              "time": "18:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 11.2,
              "summary": "Średnie zachmurzenie",
              "time": "21:00",
              "undefined": 3,
            },
            Object {
              "Temperatura (°C)": 12.7,
              "summary": "Średnie zachmurzenie",
              "time": "00:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 13.7,
              "summary": "Słaby wiatr, duże zachmurzenie",
              "time": "03:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 14.2,
              "summary": "Umiarkowany wiatr, duże zachmurzenie",
              "time": "06:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 16.5,
              "summary": "Słaby wiatr, duże zachmurzenie",
              "time": "09:00",
              "undefined": 4,
            },
            Object {
              "Temperatura (°C)": 18.9,
              "summary": "Duże zachmurzenie",
              "time": "12:00",
              "undefined": 11,
            },
            Object {
              "Temperatura (°C)": 17.7,
              "summary": "Duże zachmurzenie",
              "time": "15:00",
              "undefined": 4,
            },
            Object {
              "Temperatura (°C)": 15.5,
              "summary": "Duże zachmurzenie",
              "time": "18:00",
              "undefined": 2,
            },
            Object {
              "Temperatura (°C)": 13.6,
              "summary": "Niewielkie zachmurzenie",
              "time": "21:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 12.6,
              "summary": "Bezchmurnie",
              "time": "00:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 11.6,
              "summary": "Bezchmurnie",
              "time": "03:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 9.1,
              "summary": "Niewielkie zachmurzenie",
              "time": "06:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 11,
              "summary": "Bezchmurnie",
              "time": "09:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 15.2,
              "summary": "Bezchmurnie",
              "time": "12:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 14.8,
              "summary": "Bezchmurnie",
              "time": "15:00",
              "undefined": 0,
            },
            Object {
              "Temperatura (°C)": 11.8,
              "summary": "Bezchmurnie",
              "time": "18:00",
              "undefined": 0,
            },
          ]
        }
        layout="horizontal"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          strokeDasharray="3 3"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="time"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          stroke=""
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              "auto",
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tick={
            Object {
              "stroke": "",
            }
          }
          tickCount={5}
          type="number"
          width={60}
          yAxisId="left"
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              100,
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="right"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tick={
            Object {
              "stroke": "",
            }
          }
          tickCount={5}
          type="number"
          width={60}
          yAxisId="right"
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={
            <CustomTooltip
              tempText={
                Object {
                  "temperature": "Temperatura",
                  "temperatureMax": "Temperatura Max.",
                  "temperatureMin": "Temperatura Min.",
                  "today": "Dziś",
                  "weekDay": Array [
                    "Niedziela",
                    "Poniedziałek",
                    "Wtorek",
                    "Środa",
                    "Czwartek",
                    "Piątek",
                    "Sobota",
                    "Dziś",
                  ],
                  "weekDayShort": Array [
                    "Nie",
                    "Pon",
                    "Wt",
                    "Śr",
                    "Czw",
                    "Pt",
                    "Sob",
                  ],
                }
              }
              type="hourly"
              units="si"
            />
          }
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemSorter={[Function]}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Legend
          align="center"
          iconSize={14}
          layout="vertical"
          verticalAlign="bottom"
        />
        <Line
          activeDot={
            Object {
              "r": 8,
            }
          }
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="Temperatura (°C)"
          dot={true}
          fill="#fff"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke=""
          strokeWidth={1}
          type="monotone"
          xAxisId={0}
          yAxisId="left"
        />
        <Line
          activeDot={true}
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dot={true}
          fill="#fff"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke=""
          strokeWidth={1}
          type="monotone"
          xAxisId={0}
          yAxisId="right"
        />
      </LineChart>
    </ResponsiveContainer>
  </div>
</section>
`;
